pinot-core/src/main/java/org/apache/pinot/core/transport/server/routing/stats/ServerRoutingStatsManager.java

```diff
@@ -47,8 +48,13 @@ public class ServerRoutingStatsManager {
   private final PinotConfiguration _config;
   private volatile boolean _isEnabled;
   private ConcurrentHashMap<String, ServerRoutingStatsEntry> _serverQueryStatsMap;
+
+  // Main executor service for collecting and aggregating stats for all servers.
   private ExecutorService _executorService;
 
+  // ScheduledExecutorServer for processing periodic tasks like decay.
+  private ScheduledExecutorService _periodicTaskExecutor;
```

Comment:
Could the periodic tasks share the same executor service by changing the main executor service to a scheduled one? That way the threadpool size can still be modified w/ pinot config

Reply:
The reasoning for not using the same executorService is as follows: The main executor service is responsible for collecting stats about servers. The number of periodic tasks is proportional to the number of servers. And we wouldn't want it to interfere with stats collection - which is critical for making the routing decision.

However, we can have a separate config to control the number of periodic task threads. But I do not see us needing more than 1 thread for this purpose. Let's say, autoDecay window is 30seconds. With 200 severs, we would be generating 2*200 = 400 periodic tasks every 30 seconds. A single thread should be sufficient to process these tasks.

Thoughts?

Reply:
👍 Thanks for explaining reasoning. That makes sense to me and seems fine to leave as is.